backend/src/signinformAPI.ts

```diff
@@ -6,17 +6,26 @@ import { PermissionsManager } from './permissions';
 const checkIfDocExists = async (id: string): Promise<boolean> =>
   (await signInFormCollection.doc(id).get()).exists;
 
+export const signInFormExpired = async (id: string): Promise<boolean> => {
+  const expireAt = await (await signInFormCollection.doc(id).get()).data()?.expireAt;
+  if (expireAt === undefined) {
+    return false;
+  }
+  return expireAt <= new Date().getTime();
+};
+
 export const signInFormExists: (id: string) => Promise<boolean> = checkIfDocExists;
 
 export const createSignInForm = async (
   id: string,
+  expireAt: number,
   user: IdolMember
-): Promise<{ id: string; createdAt: number }> => {
+): Promise<{ id: string; createdAt: number; expireAt: number }> => {
   if (!PermissionsManager.canEditSignIn(user)) {
     throw new PermissionError("You don't have permission to create a sign-in form!");
   }
```

Comment:
I think it might be good to check that expiry time for the form isn't in the past here

Reply:
Done! also displaying it on frontend
![image](https://user-images.githubusercontent.com/55467524/137226432-7b4c6b00-1e5f-4a98-8fa1-a4249d41f8a7.png)